pkg/client/unversioned/thirdpartyresources.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unversioned
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+// ThirdPartyResourceNamespacer has methods to work with ThirdPartyResource resources in a namespace
+type ThirdPartyResourceNamespacer interface {
+	ThirdPartyResources(namespace string) ThirdPartyResourceInterface
+}
+
+type ThirdPartyResourceInterface interface {
+	List(label labels.Selector, field fields.Selector) (*extensions.ThirdPartyResourceList, error)
+	Get(name string) (*extensions.ThirdPartyResource, error)
+	Create(ctrl *extensions.ThirdPartyResource) (*extensions.ThirdPartyResource, error)
+	Update(ctrl *extensions.ThirdPartyResource) (*extensions.ThirdPartyResource, error)
+	UpdateStatus(ctrl *extensions.ThirdPartyResource) (*extensions.ThirdPartyResource, error)
+	Delete(name string) error
+	Watch(label labels.Selector, field fields.Selector, opts unversioned.ListOptions) (watch.Interface, error)
+}
+
+// thirdPartyResources implements DaemonsSetsNamespacer interface
+type thirdPartyResources struct {
+	r  *ExtensionsClient
+	ns string
+}
+
+func newThirdPartyResources(c *ExtensionsClient, namespace string) *thirdPartyResources {
+	return &thirdPartyResources{c, namespace}
+}
+
+// Ensure statically that thirdPartyResources implements ThirdPartyResourcesInterface.
+var _ ThirdPartyResourceInterface = &thirdPartyResources{}
+
+func (c *thirdPartyResources) List(label labels.Selector, field fields.Selector) (result *extensions.ThirdPartyResourceList, err error) {
```

Comment:
See https://github.com/kubernetes/kubernetes/pull/18075. This has changed and should only take unversioned.ListOptions now.